modules/renter/hostdb/hosttree/hosttree.go

```diff
@@ -0,0 +1,290 @@
+package hosttree
+
+import (
+	"crypto/rand"
+	"errors"
+	"sync"
+
+	"github.com/NebulousLabs/Sia/modules"
+	"github.com/NebulousLabs/Sia/types"
+)
+
+var (
+	// ErrWeightTooHeavy is returned from a Fetch() call if a weight that exceeds
+	// the total weight of the tree is requested.
+	ErrWeightTooHeavy = errors.New("requested a too-heavy weight")
+
+	// ErrNegativeWeight is returned from an Insert() call if an entry with a
+	// negative weight is added to the tree. Entries must always have a positive
+	// weight.
+	ErrNegativeWeight = errors.New("cannot insert using a negative weight")
+
+	// ErrNilEntry is returned if a fetch call results in a nil tree entry. nodes
+	// should always have a non-nil entry, unless they have been Delete()ed.
+	ErrNilEntry = errors.New("node has a nil entry")
+
+	// ErrHostExists is returned if an Insert is called with a public key that
+	// already exists in the tree.
+	ErrHostExists = errors.New("host already exists in the tree")
+
+	// ErrNoSuchHost is returned if Remove is called with a public key that does
+	// not exist in the tree.
+	ErrNoSuchHost = errors.New("no host with specified public key")
+)
+
+type (
+	// WeightFunc is a function used to weight a given HostDBEntry in the tree.
+	WeightFunc func(modules.HostDBEntry) types.Currency
+
+	// HostTree is a data structure that contains a weighted tree of
+	// `HostEntries`, and provides methods for inserting, retreiving, and
+	// modifing entries.
+	HostTree struct {
```

Comment:
I usually leave out the `''` unless I'm referring to a variable name that isn't easy to parse. The standard library doesn't use them at all. For example:
```go
// Abs returns the absolute value of x.
func Abs(x float64) float64
```
Here it's pretty obvious what `x` is referring to, so you don't need any special marks.
```go
// Abs returns an absolute representation of path.
func Abs(path string) string
```
Here it's a little less obvious, because `path` doesn't stick out as much as `x`. Still, it's passable.

Types, especially exported ones, stand out pretty clearly so they can be typed verbatim. You just need to use your best judgment to decide whether a name will confuse the reader. For example, in docstring of `Fetch`:
```go
// Hosts passed to `ignore` will not be considered.
func (ht *HostTree) Fetch(n int, ignore []types.SiaPublicKey)
```
Here, it would be confusing if `ignore` weren't quoted somehow. So surrounding it in single quotes is recommended.